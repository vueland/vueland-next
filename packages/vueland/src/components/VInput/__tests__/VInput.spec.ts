import { mount } from '@vue/test-utils'
import { VInput } from '../index'

describe('EInput', () => {
  const mountFunction = (options = {}) => mount(VInput, {
    ...options
  })

  it('should mount component and match snapshot', () => {
    const wrap = mountFunction()

    expect(wrap.html()).toMatchSnapshot()
  })

  it('should test "disabled" prop', () => {
    const wrapper = mountFunction({
      props: { disabled: true }
    })

    expect(wrapper.classes()).toContain('v-input--disabled')
  })

  // it('should test "focused" prop', () => {
  //   const wrapper = mountFunction({
  //     props: { focused: true }
  //   })
  //
  //   expect(wrapper.classes()).toContain('v-input--focused')
  // })

  it('should test "label"', () => {
    const wrapper = mountFunction({
      props: { label: 'test label' }
    })

    expect(wrapper.find('label').text()).toContain('test label')
  })

  it('should test "color" prop with color class name', () => {
    const wrapper = mountFunction({
      props: { color: 'grey darken-1' }
    })

    expect(wrapper.attributes('class')).toContain('grey--text text--darken-1')
  })

  it('should test "color" prop with css color', () => {
    const wrapper = mountFunction({
      props: { color: '#fa5a5a' }
    })

    expect(wrapper.attributes().style).toContain('color: rgb(250, 90, 90);')
  })

  it('should test css "color" if "disabled" is true', () => {
    const wrapper = mountFunction({
      props: { color: 'orange darken-1', disabled: true }
    })

    expect(
      wrapper.attributes().class.includes('orange--text text--darken-1')
    ).toBeFalsy()
  })

  it('should test prepend-icon prop', () => {
    const wrapper = mountFunction({
      props: { prependIcon: 'fas fa-check' }
    })

    expect(wrapper.find('.v-input--has-prepend-icon').exists()).toBe(true)
  })

  it('should test prepend-icon slot', () => {
    const wrapper = mountFunction({
      slots: {
        ['prepend-icon']: `<i class="fas fa-check"></i>`
      }
    })

    expect(wrapper.find('.v-input--has-prepend-icon').exists()).toBe(true)
  })

  it('should test append-icon prop', () => {
    const wrapper = mountFunction({
      props: { appendIcon: 'fas fa-check' }
    })

    expect(wrapper.find('.v-input--has-append-icon').exists()).toBe(true)
  })

  it('should test append-icon slot', () => {
    const wrapper = mountFunction({
      slots: {
        ['append-icon']: `<i class="fas fa-check"></i>`
      }
    })

    expect(wrapper.find('.v-input--has-append-icon').exists()).toBe(true)
  })

  it('should test "text-color" for "text-field" slot', () => {
    const textField = `
      <template #text-field="{textCssColor, textClassColor}">
        <div  :class="['test', textClassColor]" :style="textCssColor"></div>
      </template>
    `
    const wrapper = mountFunction({
      props: { textColor: '#fa5a5a' },
      slots: {
        ['text-field']: textField
      }
    })

    expect(wrapper.find('.test').attributes('style')).toContain(
      'color: rgb(250, 90, 90)'
    )
  })
})
