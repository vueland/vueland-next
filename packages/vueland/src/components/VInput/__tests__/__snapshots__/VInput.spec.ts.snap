// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EInput should mount component and match snapshot 1`] = `
<div class="v-input">
  <div class="v-input__label">
    <label class="v-label v-label--on-input">
    </label>
  </div>
  <div class="v-input__text-field">
  </div>
  <div class="v-input__hints">
    <transition-stub name="fade"
                     mode
                     appear="false"
                     persisted="false"
                     css="true"
    >
    </transition-stub>
  </div>
</div>
`;
